classes/blocks/class-articles.php

```diff
@@ -314,17 +314,25 @@ function ( $tag ) {
 		// Get all posts with arguments.
 		$args = self::DEFAULT_POST_ARGS;
 
+		// For post page block, so current main post will exclude.
+		if ( $exclude_post_id ) {
+			$args['post__not_in'] = [ $exclude_post_id ];
+
+			// Get current post tags & categories, if not pass.
+			if ( !$tags ) {
```

Comment:
These are the new code changes after last review.
Reference : https://github.com/greenpeace/planet4-master-theme/pull/1645/files#r823477472

cc: @comzeradd